### PR TITLE
Fix proto tooling

### PR DESCRIPTION
**Pull request by arkodg**
* Add `protoc-gen-go-grpc` & `protoc-gen-go` dep to `make protos`

* init go mod for `protoc-gen-go`

### CODE REVIEW

File: tools/src/protoc-gen-go/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/envoyproxy/gateway/tools/src/protoc-gen-go
```
Comment (arkodg) [inline tools/src/protoc-gen-go/go.mod:1]: hey @AliceProxy any idea why this wasn't added in the orig PR
https://github.com/envoyproxy/gateway/pull/1209/

Comment (arkodg) [reply tools/src/protoc-gen-go/go.mod:1]: cc @haq204

Comment (AliceProxy) [reply tools/src/protoc-gen-go/go.mod:1]: Huh... I remember seeing that when I was messing around with stuff... maybe I forgot to push it or something when I was trying to figure out a cleaner way to handle the buf files?

**Review by AliceProxy — DISMISSED** [review]

**Comment by ZackButcher** [comment]
@arkodg I think you should also update `buf.gen.yaml` in this PR to output the `go` files in a relative dir as well:
```yaml
version: v1
plugins:
  - plugin: go
    out: ./
    path: tools/bin/protoc-gen-go
    opt: paths=source_relative
  - plugin: go-grpc
    out: ./
    path: tools/bin/protoc-gen-go-grpc
    opt: paths=source_relative
```

Right now it's only set for gRPC.

File: tools/src/protoc-gen-go/go.mod (hunk 1 above)
Comment (arkodg) [reply tools/src/protoc-gen-go/go.mod:1]: cool thanks for confirming !

**Comment by arkodg** [comment]
> @arkodg I think you should also update `buf.gen.yaml` in this PR to output the `go` files in a relative dir as well:
> 
> ```yaml
> version: v1
> plugins:
>   - plugin: go
>     out: ./
>     path: tools/bin/protoc-gen-go
>     opt: paths=source_relative
>   - plugin: go-grpc
>     out: ./
>     path: tools/bin/protoc-gen-go-grpc
>     opt: paths=source_relative
> ```
> 
> Right now it's only set for gRPC.

thanks @ZackButcher , done

**Review by AliceProxy — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Pull request merged**